docs/build-results-archive.md

```diff
@@ -0,0 +1,52 @@
+# Build results archive.
+
+This document describes a build results archive format which can be used to store the sarif results of a build-and-test procedure.
+
+The archive will take the form of a compressed (zstd if available, or bzip2 otherwise) tar file containing the following files.
+
+```
+./build-results-archive
+./build/**/*.sarif
+./colcon-build-cmd
+./colcon-test-cmd
+./processed.sarif
+./vcs-export-exact.repos
+```
+
+The individual files are further described:
+
+## build-results-archive
+
+This is a JSON file containing metadata regarding the current archive.
+The file _must_ contain an single top-level JSON object and that object _must_ contain the `"version"` key with the value of `1`.
+Future revisions of this format may add additional valid versions.
+
+## build/**/*.sarif
+
+The build/ subdirectory _must_ contain the unaltered sarif content of the workspace build directory after the end of the latest colcon test run.
+
+## colcon-build-cmd
+
+This file _must_ contain the full colcon build command described in the colcon logs under the `latest_build` symlink.
+The file, when run as a /bin/sh program or prefixed with the `exec` shell command in a workspace root should invoke colcon in the local environment using the same arguments as the original build.
+
+## colcon-test-cmd
+
+This file is identical in behavior to `colcon-build-cmd` but should describe the test command under the `latest_test` symlink.
+
+## processed-sarif-cmd
```

Comment:
Once we have a "colcon sarif" command (whatever we decide the specific verb name should be), this could become "colcon-sarif-cmd", and would then be unform with "colcon-build-cmd", and "colcon-test-cmd".

Reply:
It could although I'm not entirely convinced that it should be a colcon command or that even if it is a colcon command that it needs to be and therefore semantically I would argue against putting colcon in the filename. In fact I'd probably sooner argue that we should strip it from the build and test command filenames and call them `build-cmd` and `test-cmd` except that there really isn't a substitute for colcon in those cases and I think it's worth separating that definition from the build and test commands for any one package.

If we want to bikeshed a little perhaps `workspace-{build,test}-cmd`?